app/scripts/modules/core/src/manifest/stage/JobStageExecutionLogs.tsx

```diff
@@ -15,77 +15,25 @@ interface IJobStageExecutionLogsProps {
   application: Application;
   externalLink: string;
   podNameProvider: IPodNameProvider;
+  location: string;
 }
 
 interface IJobStageExecutionLogsState {
-  subscription: IManifestSubscription;
-  manifestId: string;
+  manifest?: IManifest;
 }
 
 export class JobStageExecutionLogs extends React.Component<IJobStageExecutionLogsProps, IJobStageExecutionLogsState> {
   public state = {
-    subscription: { id: '', unsubscribe: () => {}, manifest: {} } as IManifestSubscription,
-    manifestId: '',
+    manifest: {} as IManifest,
   };
 
   public componentDidMount() {
-    this.componentDidUpdate(this.props, this.state);
-  }
-
-  public componentWillUnmount() {
-    this.unsubscribe();
-  }
-
-  private unsubscribe() {
-    this.state.subscription && this.state.subscription.unsubscribe && this.state.subscription.unsubscribe();
-  }
-
-  public componentDidUpdate(_prevPropds: IJobStageExecutionLogsProps, prevState: IJobStageExecutionLogsState) {
-    const { manifest } = this.props;
-    const manifestId = ManifestService.manifestIdentifier(manifest);
-    if (prevState.manifestId === manifestId) {
-      return;
-    }
-    this.refreshSubscription(manifestId, manifest);
-  }
-
-  private refreshSubscription(manifestId: string, manifest: IStageManifest) {
-    const subscription = {
-      id: manifestId,
-      manifest: this.stageManifestToIManifest(manifest, this.props.deployedName, this.props.account),
-      unsubscribe: this.subscribeToManifestUpdates(manifest),
-    };
-    this.setState({ subscription, manifestId });
-  }
-
-  private subscribeToManifestUpdates(manifest: IStageManifest): () => void {
-    const params = {
-      account: this.props.account,
-      name: this.props.deployedName,
-      location: manifest.metadata.namespace == null ? '_' : manifest.metadata.namespace,
-    };
-    return ManifestService.subscribe(this.props.application, params, (updated: IManifest) => {
-      const subscription = { ...this.state.subscription, manifest: updated };
-      this.setState({ subscription });
+    const { account, location, deployedName } = this.props;
+    ManifestReader.getManifest(account, location, deployedName).then(manifest => {
+      this.setState({ manifest });
```

Comment:
perhaps use the cancelable promise pattern here? https://github.com/spinnaker/deck/wiki/Conventions#setting-state-in-promises